test/Canvas.test.ts

```diff
@@ -0,0 +1,1330 @@
+        )
+      )()
+
+      // Actual
+      testCtx.beginPath()
+      testCtx.clip(path, fillRule)
+
+      assertCalledWith(ctx.clip as jest.Mock, path, fillRule)
+
+      assert.deepStrictEqual(ctx.__getDrawCalls(), testCtx.__getDrawCalls())
+    })
+  })
+
+  describe('lineTo', () => {
+    it('should draw a line to the specified point', () => {
+      const p = S.point(10, 10)
+
+      // Test
+      render(
+        pipe(
+          C.beginPath,
+          R.chain(() => C.lineTo(p)),
+          R.chain(() => C.stroke())
+        )
+      )()
+
+      // Actual
+      testCtx.beginPath()
+      testCtx.lineTo(p.x, p.y)
+      testCtx.stroke()
+
+      assertCalledWith(ctx.lineTo as jest.Mock, 10, 10)
+
+      assert.deepStrictEqual(ctx.__getDrawCalls(), testCtx.__getDrawCalls())
+    })
+  })
+
+  describe('moveTo', () => {
+    it('should draw a line to the specified point', () => {
+      const p = S.point(10, 10)
+
+      // Test
+      render(
+        pipe(
+          C.beginPath,
+          R.chain(() => C.moveTo(p)),
+          R.chain(() => C.stroke())
+        )
+      )()
+
+      // Actual
+      testCtx.beginPath()
+      testCtx.moveTo(p.x, p.y)
+      testCtx.stroke()
+
+      assertCalledWith(ctx.moveTo as jest.Mock, 10, 10)
+
+      assert.deepStrictEqual(ctx.__getDrawCalls(), testCtx.__getDrawCalls())
+    })
+  })
+
+  describe('closePath', () => {
+    it('should close the current path', () => {
+      // Test
+      render(
+        pipe(
+          C.beginPath,
+          R.chain(() => C.stroke()),
+          R.chain(() => C.closePath)
+        )
+      )()
+
+      // Actual
+      testCtx.beginPath()
+      testCtx.stroke()
+      testCtx.closePath()
+
+      assertCalledWith(ctx.closePath as jest.Mock)
+
+      assert.deepStrictEqual(ctx.__getDrawCalls(), testCtx.__getDrawCalls())
+    })
+  })
+
+  describe('strokePath', () => {
+    it('should begin and stroke a path', () => {
+      const color = pipe(Color.white, Color.toCss)
+      const first = S.point(20, 20)
+      const second = S.point(10, 10)
+
+      render(
+        C.strokePath(
+          pipe(
+            C.setStrokeStyle(color),
+            R.chain(() => C.moveTo(first)),
+            R.chain(() => C.lineTo(second))
+          )
+        )
+      )()
+
+      // Actual
+      testCtx.beginPath()
+      testCtx.strokeStyle = color
+      testCtx.moveTo(first.x, first.y)
+      testCtx.lineTo(second.x, second.y)
+      testCtx.stroke()
+
+      assert.deepStrictEqual(ctx.__getDrawCalls(), testCtx.__getDrawCalls())
+    })
+  })
+
+  describe('fillPath', () => {
+    it('should begin and stroke a path', () => {
+      const color = pipe(Color.black, Color.toCss)
+      const first = S.point(20, 20)
+      const second = S.point(10, 10)
+
+      // Test
+      pipe(
+        canvas,
+        C.getContext2D,
+        IO.chain(C.fillPath(flow(C.setFillStyle(color), IO.chain(C.moveTo(first)), IO.chain(C.lineTo(second)))))
+      )()
```

Comment:
this should be

```ts
render(
  C.fillPath(
    pipe(
      C.setFillStyle(color),
      R.chain(() => C.moveTo(first)),
      R.chain(() => C.lineTo(second))
    )
  )
)()
```

I guess

Reply:
Apologies, missed that one on the initial pass. See https://github.com/gcanti/graphics-ts/pull/7/commits/c71374969b11532dbc227cf7f069517546a0214d

Reply:
NP